src/slskd/Core/Data/Migrations/Z04012025_TransferStateMigration.cs

```diff
@@ -0,0 +1,208 @@
+    {
+        using var transaction = connection.BeginTransaction();
+
+        try
+        {
+            /*
+                rename old table so we can create the new one with the layout we want
+                while preserving the existing data
+            */
+            var rename = @"
+                ALTER TABLE Transfers RENAME TO Transfers_old;
+            ";
+
+            using var renameCommand = new SqliteCommand(rename, connection, transaction);
+            renameCommand.ExecuteNonQuery();
+
+            /*
+                create the new table with StateDescription appearing after State
+            */
+            var create = @"
+                CREATE TABLE Transfers (
+                    Id TEXT NOT NULL,
+                    Username TEXT,
+                    Direction TEXT NOT NULL,
+                    Filename TEXT,
+                    Size INTEGER NOT NULL,
+                    StartOffset INTEGER NOT NULL,
+                    State INTEGER NOT NULL,
+                    StateDescription TEXT, -- new column
+                    RequestedAt TEXT NOT NULL,
+                    EnqueuedAt TEXT,
+                    StartedAt TEXT,
+                    EndedAt TEXT,
+                    BytesTransferred INTEGER NOT NULL,
+                    AverageSpeed REAL NOT NULL,
+                    PlaceInQueue INTEGER,
+                    Exception TEXT,
+                    Removed INTEGER NOT NULL,
+                    CONSTRAINT PK_Transfers PRIMARY KEY(Id)
+                );
+            ";
+
+            using var createCommand = new SqliteCommand(create, connection, transaction);
+            createCommand.ExecuteNonQuery();
+
+            Log.Information("Schema changes applied, copying data...");
+
+            /*
+                copy the existing data into the new table
+            */
+            var copy = @"
+                INSERT INTO Transfers (
+                    Id, Username, Direction, Filename, Size, StartOffset, State, 
+                    RequestedAt, EnqueuedAt, StartedAt, EndedAt, BytesTransferred, 
+                    AverageSpeed, PlaceInQueue, Exception, Removed
+                )
+                SELECT 
+                    Id, Username, Direction, Filename, Size, StartOffset, State, 
+                    RequestedAt, EnqueuedAt, StartedAt, EndedAt, BytesTransferred, 
+                    AverageSpeed, PlaceInQueue, Exception, Removed
+                FROM Transfers_old;";
+
+            using var copyCommand = new SqliteCommand(copy, connection, transaction);
+            copyCommand.ExecuteNonQuery();
+
+            Log.Information("Data copied, adjusting columns...");
+
+            /*
+                we no longer need the old table, so drop it
+            */
+            using var dropCommand = new SqliteCommand("DROP TABLE Transfers_old;", connection, transaction);
+            dropCommand.ExecuteNonQuery();
```

Comment:
suggestion: Add a column to Transfers instead of replacing the whole table.

note: SQLite's support of transactional DDL limits the corner cases that could crop up

Reply:
the new table is so that the `State` and `StateDescription` columns will be next to each other.  i'm thinking maybe it's not worth the extra effort (and file i/o)